### ISSUE TITLE
No Kerberos credentials available

### ISSUE THREAD

**Issue by rezaa1** [freeipa/ansible-freeipa#374]
ipa-server-4.6.6-11.el7.centos.x86_64
freeipa collection version 0.1.12
------
 
The playbook works with some users, and not working for other users!

This is a task i ean:
- name: add host to  hostgroup
  freeipa.ansible_freeipa.ipahostgroup:
    ipaadmin_password: "{{ipaadmin_password}}"
    name: "{{host_location}}"
    action: member
    host: "{{host_fqdn}}"

-------

The full traceback is:
WARNING: The below traceback may *not* be related to the actual failure.
  File "/tmp/ansible_freeipa.ansible_freeipa.ipahostgroup_payload_5z8Agp/ansible_freeipa.ansible_freeipa.ipahostgroup_payload.zip/ansible_collections/freeipa/ansible_freeipa/plugins/modules/ipahostgroup.py", line 240, in main
  File "/tmp/ansible_freeipa.ansible_freeipa.ipahostgroup_payload_5z8Agp/ansible_freeipa.ansible_freeipa.ipahostgroup_payload.zip/ansible_collections/freeipa/ansible_freeipa/plugins/module_utils/ansible_freeipa_module.py", line 156, in api_connect
    backend.connect()
  File "/usr/lib/python2.7/site-packages/ipalib/backend.py", line 69, in connect
    conn = self.create_connection(*args, **kw)
  File "/usr/lib/python2.7/site-packages/ipaserver/plugins/ldap2.py", line 196, in create_connection
    principal = krb_utils.get_principal(ccache_name=ccache)
  File "/usr/lib/python2.7/site-packages/ipalib/krb_utils.py", line 171, in get_principal
    raise errors.CCacheError(message=unicode(e))
fatal: [lab-01.ipa.example.com]: FAILED! => {
    "changed": false, 
    "invocation": {
        "module_args": {
            "action": "member", 
            "description": null, 
            "host": [
                "rqtest-01.lab.example.com"
            ], 
            "hostgroup": null, 
            "ipaadmin_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "ipaadmin_principal": "admin", 
            "name": [
                "melpop_lab"
            ], 
            "nomembers": null, 
            "state": "present"
        }
    }, 
    "msg": "Major (851968): Unspecified GSS failure.  Minor code may provide more information, Minor (2529639053): No Kerberos credentials available (default cache: KEYRING:persistent:1604100503)"
}

**Comment by rjeffman** [comment]
@rezaa1, can you run the CLI command `ipa hostgroup-add-member` with the users that are failing to execute this playbook?

I usually see this error with expired passwords.

**Comment by rezaa1** [comment]
The password has is not expired i can login to servers with that user. but that user is service user which has no access, i use admin credential with the playbook to run the command

Here you see the same error when i try the command line.
![image](https://user-images.githubusercontent.com/18672784/91374962-c3553d00-e85c-11ea-8561-0d93f1377441.png)

**Comment by rjeffman** [comment]
That behaviour is expected if you did not run `kinit` before the commands.

I'd like to ask you a few things so we can find what is wrong.

Log into the machine, run `kinit admin`, run `klist -A`, run `ipa hostgroup-add-member`. Do NOT log out, or run `kdestroy`.

Back in the controller, try to run the playbook that failed before. If you have any errors, check the `kinit` you find in /var/log/krb5kdc.log.

**Comment by rezaa1** [comment]
yes that works this way, but i am running this module with admin credntial and expect to be able to run it, my kerberose ticket will expire after 24h so you mean each time i need to run this command i have to go on the server and run kinit? That doesn't look good.

**Comment by t-woerner** [comment]
@rezaa1 Please explain what "The playbook works with some users, and not working for other users!" means. Which users are these?
The use of kinit is needed for the command line `ipa hostgroup-add-member`, but not for the ansible-freeipa module. The module is using kinit internally.

**Comment by rezaa1** [comment]
for instance my normal user which I always login as works, but this service account which we only use it for tower to login and do tasks, it won't work.
The issue is , as soon as we login to the server and then su - ansible, then kinit admin the playbook works too. but when we do kdestroy in command line , it fails!

**Comment by t-woerner** [comment]
Do you have environment variables `KRB5CCNAME` or `KRB5_CLIENT_KTNAME` set?

**Comment by rezaa1** [comment]
yes i have the variable set, but it doesn't do anything. if you do not have valid Kerberos ticket.

**Comment by t-woerner** [comment]
These variables are used by the modules. Where have you set these variables? For the service user locally or in the ansible environment? Please verify that the settings of the variables are correct. Are you using `become_method: ksu`?

**Comment by rezaa1** [comment]
I use become_method: sudo, the only way i could get it working was to set become: false

**Comment by mproehl** [comment]
Why should it be necessary to set KRB5CCNAME when Kerberos credentials are available in the default credential cache?
